spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/BlockingQueueConsumer.java

```diff
@@ -794,15 +798,13 @@ public boolean commitIfNecessary(boolean locallyTransacted) throws IOException {
 
 			boolean ackRequired = !this.acknowledgeMode.isAutoAck() && !this.acknowledgeMode.isManual();
 
-			if (ackRequired) {
-				if (!this.transactional || isLocallyTransacted) {
-					long deliveryTag = new ArrayList<Long>(this.deliveryTags).get(this.deliveryTags.size() - 1);
-					this.channel.basicAck(deliveryTag, true);
-				}
+			if (ackRequired && (!this.transactional || isLocallyTransacted)) {
+				long deliveryTag = new ArrayList<Long>(this.deliveryTags).get(this.deliveryTags.size() - 1);
```

Comment:
Why do we need to wrap into a new list to get the last one item anyway?

Reply:
`this.deliveryTags` is a `LinkedHashSet` - there is no easy way to get the last element.